src/main/java/de/torui/coflsky/proxy/ProxyManager.java

```diff
@@ -0,0 +1,99 @@
+package de.torui.coflsky.proxy;
+
+import de.torui.coflsky.CoflSky;
+import de.torui.coflsky.commands.models.ProxyRequest;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class ProxyManager {
+    private final String ProxyResponseUrl = "http://sky.coflnet.com/api/data/proxy";
+    private final ExecutorService requestExecutor = Executors.newSingleThreadExecutor();
+
+
+    public void handleRequestAsync(ProxyRequest request){
+        if(request.isUploadEnabled()) {
+            CompletableFuture<String> req = this.doRequest(request.getUrl());
+            req.thenAcceptAsync(res -> this.uploadData(res,request.getId()));
+        }
```

Comment:
```suggestion
         CompletableFuture<String> req = this.doRequest(request.getUrl());
        if(request.isUploadEnabled()) {
            req.thenAcceptAsync(res -> this.uploadData(res,request.getId()));
        }
```
The request should be made when its requested just not uploaded